core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -1165,8 +1165,22 @@ private void substituteSubQuery(Blackboard bb, SubQuery subQuery) {
       if (!config.isExpand()) {
         return;
       }
-      converted = convertExists(query, RelOptUtil.SubQueryType.EXISTS,
-          subQuery.logic, true, null);
+      final SqlValidatorScope seekScope =
+          (query instanceof SqlSelect)
+              ? validator.getSelectScope((SqlSelect) query)
+              : null;
+      final Blackboard seekBb = createBlackboard(seekScope, null, false);
+      final RelNode seekRel = convertQueryOrInList(seekBb, query, null);
+      // An existential sub-query whose inner child has at least 1 tuple
+      // (e.g. an Aggregate with no grouping columns or non-empty Values node)
+      // should be simplified to a Boolean constant expression.
+      final RelMetadataQuery mq = seekRel.getCluster().getMetadataQuery();
+      if (mq.getMinRowCount(seekRel) > 0.0) {
+        subQuery.expr = rexBuilder.makeLiteral(true);
```

Comment:
Should we introduce statistics directly during sqlNode to RelNode converting ? Some how we can not make sure the statistics is definitely right, this may causes wrong results. I think it's a little radical promotion.

Reply:
`getMinRowCount` doesn't utilize statistics. Instead, it just return the minimum row count of the relation, which is guaranteed. You can check the implementation of its metadata handler.

Reply:
@hsyuan As the java doc has presented:
>Estimates the minimum number of rows which will be returned by a
>relational expression.
>
>The default implementation for this query returns 0,
>but metadata providers can override this with their own cost models

This value defaults to be 0 and is estimated, so it is not 100% accurate. I still think we should use statistics 
 metadata to auxiliary planning but not deterministic node rewrite.